<a href="https://colab.research.google.com/github/ChitikelaNeelaPadmavathi/python-codes/blob/main/CODES.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#1.PALINDROME

In [ ]:
s=int(input ())
reverse=int(str(s)[::-1])
if s==reverse:
    print ("palindrome")
else:
    print ("not a palindrome ")

34567
not a palindrome 


In [ ]:

def is_palindrome(string):
    string = string.lower().replace(" ", "")  # Convert to lowercase and remove spaces
    return string == string[::-1]  # Check if the processed string is the same when reversed

# Get user input
s = input("Enter a string: ")

# Check and print the result
if is_palindrome(s):
    print("Palindrome")
else:
    print("Not a palindrome")

Enter a string: Gdrhj
Not a palindrome


**2.AMSTRONG NUMBER**


In [ ]:
num=int(input())
digits=str(num)
num_digit=len(digits)
amstrong_sum=0
for i in digits:
  amstrong_sum+=int(i)**num_digit
if amstrong_sum==num:
    print("It is Amstrong")
else:
    print("It is not a amstrong number")



153
It is Amstrong


In [ ]:
n=int(input())
m=n
sum=0
while m!=0:
  d=m%10
  sum=sum+(d**3)
  m=m//10
if(sum==n):
  print("amstrong")
else:
  print("not amstrong")


5667
not amstrong


**3.FIBONACCI SERIES**

In [ ]:
num=int(input())
n1,n2=0,1
print("fibanacii:",n1,n2,end=" ")
for i in range(2,num):
  n3=n1+n2
  n1=n2
  n2=n3
  print(n3,end=" ")

10
fibanacii: 0 1 1 2 3 5 8 13 21 34 

**4.FACTORIAL NUMBER**

In [ ]:
num=int(input())
fact=1
for i in range(1,num+1):
  fact=fact*i
print(fact)

5
120


**5.POWER OF A NUMBER**

In [ ]:
num=int(input())
b=num*2
print(b)

5
10


**6.FACTORS OF A NUMBER**

In [ ]:
num=int(input())
for i in range(1,num+1):
  if num%i==0:
    print(i,end=" ")





8
1 2 4 8 

**7.SUM OF PRIME NUMBERS**

In [ ]:
num = int(input("enter the num: "))
sum = 0
for i in range(1,10):
     if num % i == 0:
          sum = sum + i
print(sum)

enter the num: 10
8


**8.STRONG NUMBER**
Input : 145
Output : It's a Strong Number.
Explanation : Number = 145
145 = 1! + 4! + 5!
145 = 1 + 24 + 120

In [ ]:
import math
n=int(input())
m=n
sum=0
while m!=0:
  d=m%10
  sum=sum+(math.factorial(d))
  m=m//10
if(sum==n):
  print("strorng")
else:
  print("not strong")



145
strorng


****

**9.PERFECT NUMBER**
Input : 28
Divisors : [1, 2, 4, 7, 14]
Sum = 1 + 2 + 4 + 7 + 14 = 28
Output : It's a Perfect Number


In [9]:
n = int(input())
sum = 0
for i in range(1, n):
    if n % i == 0:
        sum = sum + i
if sum == n:
    print(f"{n} is a Perfect number")
else:
    print(f"{n} is not a Perfect number")

28
28 is a Perfect number


**10 AUTOMORPHIC NUMBER**
Explanation : Number = 5
Square of number = 25
as the square of the number ends with the number itself, It's an Automorphic number.

In [17]:
num=int(input())
b=num*num
if str(b).endswith(str(num)):
  print("automorphic")
else:
    print("not automorphic")


4
not automorphic


**11.HAESHAD NUMBER**
Example
Input : 21
Output : Yes ' It's a Harshad Number.
Explanation : The sum of the digits of 21 is 3 i.e 2 + 1. As the number 21 is divisible by 3, It's a Harshad Number.

In [25]:
n =int(input())
p=n
sum1=0
while(n>0):
    sum1+=n%10
    n=n//10
if(p%sum1==0):
    print("Harshad number")
else:
    print("Not harshad number")

21
Harshad number


**12ABUNDANT NUMBER**
The Factors for the number 12 are, 1, 2, 3, 4 and 6. We don't want to include the number itself.
Now the sum of the factors except the number itself is :
1 + 2 + 3 + 4 + 6 = 16
as the number 16>12 , the number itself.
It's an abundant number.

In [30]:
n=int(input())
sum=0
for i in range(1,n):
  if n%i==0:
    sum+=i
if sum>n:
  print("abundant")
else:
  print("not abundant")

12
abundant


**13.FRIENDLY PAIR**
Example
Input : 6 28
Output : Yes, they are a friendly pair
Explanation : The factors of 6 and 28 except the numbers themselves are 1, 2, 3 and 1, 2, 4, 7, 14 respectively.
Now the sum of factors of both the numbers are 6 and 28 respectively.
When we divide the sums with the numbers we get 1 and 1 respectively.
As the ratio of both the number match, they are considered as a friendly pair.

In [48]:
# Define the two numbers
num1 = 6
num2 = 28

# Initialize sums of proper divisors
sum1 = 0
sum2 = 0

# Calculate the sum of proper divisors of num1
for i in range(1, num1):
    if num1 % i == 0:
        sum1 += i

# Calculate the sum of proper divisors of num2
for i in range(1, num2):
    if num2 % i == 0:
        sum2 += i

# Print sums for debugging
print(f"Sum of proper divisors of {num1}: {sum1}")
print(f"Sum of proper divisors of {num2}: {sum2}")

# Determine if the numbers are friendly pairs (amicable pairs)
if sum1 == num2 and sum2 == num1:
    print("Friendly pair")
else:
    print("Not a friendly pair")



Sum of proper divisors of 6: 6
Sum of proper divisors of 28: 28
Not a friendly pair
